### PR TITLE
Deduplicate test kernel for permission tests

### PR DESCRIPTION
**Pull request by markusthoemmes**
These tests have a lot in common, so it might come in handy to deduplicate their kernels and types to avoid divergence.

### CODE REVIEW

File: test/user_permissions.go
Comment (matzew) [inline test/user_permissions.go:43]: I guess parameterizing is not worth it - nor needed, right?

**Review by matzew — APPROVED** [review]
/lgtm
/approve

nice refactors here, thanks!

**Comment by markusthoemmes** [comment]
/hold

#1119 should land first imo.

File: test/user_permissions.go
Comment (matzew) [inline test/user_permissions.go:57]: you will include the `PollImmediate` section here too, right?

Comment (markusthoemmes) [reply test/user_permissions.go:57]: after the other PR merges, yep

Comment (matzew) [reply test/user_permissions.go:57]: cool, thanks

File: test/user_permissions.go
Comment (markusthoemmes) [reply test/user_permissions.go:43]: it's the same everywhere currently, yep. I'd parametrize if needed.

File: test/user_permissions.go
Comment (markusthoemmes) [reply test/user_permissions.go:58]: Done.

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Comment by matzew** [comment]
hrm...

/test 4.7-operator-e2e-aws-ocp-47

**Pull request merged**
